File: 1/problem_one.py
#!/usr/bin/env python

from solution import TaxiMap


def main():
    taxi_map = TaxiMap()
    with open('input', 'rb') as handle:
        for instruction in map(str.strip, handle.read().split(',')):
            taxi_map.step(instruction)
        print "Blocks Away:", taxi_map.blocks_away()

if __name__ == '__main__':
    main()

File: 1/nim/problem_one.nim
from strutils import split, strip
from taximap import TaxiMap, newTaxiMap, step, blocks_away 

var
  f: File

if open(f, "../input"):
  var
    map: TaxiMap = newTaxiMap()

  let line = readLine(f)
  let instructions = split(line, ',')

  for instruction in instructions:
    map.step(strip(instruction))

  echo "Blocks Away: ", map.blocks_away()
else:
  echo "Couldn't open input file."
